### ISSUE TITLE
Version control: Update via Meld directly between versions not only complete replace

### ISSUE THREAD

**Issue by Phaelixx** [zim-desktop-wiki/zim-desktop-wiki#1539]
Hi there,
I really like ZIM and have recently discovered the "Version Control" Plugin. It works well, but I am struggling with merging of different versions.
If I have two different versions of the same note, I can either replace the note with one revision or make changes manually.

I think there could be an easy improvement to this plugin. If I am opening the one revision via "Side by Side" on Linux "Meld" is opened and I can compare the current note with the revision. I'm able **to choose which single lines or paragraphs** I want to put into the current note.
Unfortunately this is not saved to the ZIM Note.
Instead there is a folder "/tmp/zim-xxx" there the revisions and an additional File with the content of the current version are stored. Only this file will be updated.
If instead of the "dummy" file the path of the original note would be passed to meld, that issue should be solved.

Can you make it possible to update the file in the original notebook folder and not the "dummy" file in the temporary folder? 

I would really appreciate it, since it would save me and possible other people a lot of time of messy copy&paste:) 

Best regards,
Felix
PS: Thanks for the amazing work with ZIM!

**Comment by stho** [comment]
This could be done by normal git usage. Do you know how it works?

It is important that you have committed all changes before merging it. 

In order to achieve that you need a remote git repository for your local git repo of your notes. You can get a remote repo here at GitHub or any other provider.

**Comment by Phaelixx** [comment]
No, I never used it before. I just gave it a try and it seemed to work fine apart from the mentioned issue.

Thank you for the hints, I will look into it!

**Comment by introt** [comment]
Hi @Phaelixx, does "Plugin for detecting and merging changes on disk #1986" work for your use case?
